core/jobs/batch_jobs/story_node_jobs.py

```diff
@@ -0,0 +1,183 @@
+                                break
+                        if story_published_on is not None:
+                            break
+
+                    current_story_version = story_model.version
+                    node_created_on = None
+                    for version in range(current_story_version, 0, -1):
+                        snapshot_id = story_model.get_snapshot_id(
+                                story_model.id, version)
+                        story_metadata = (
+                            story_models.StorySnapshotMetadataModel.get(
+                                snapshot_id))
+                        for cmd in story_metadata.commit_cmds:
+                            if (cmd['cmd'] == 'update_story_node_property' and
+                                cmd['node_id'] == node['id'] and
+                                node.get('last_modified_msecs') is None):
+                                node['last_modified_msecs'] = (
+                                    utils.get_time_in_millisecs(
+                                    story_metadata.created_on))
+
+                            if (cmd['cmd'] == 'add_story_node' and
+                                cmd['node_id'] == node['id']):
+                                node_created_on = (
+                                    utils.get_time_in_millisecs(
+                                    story_metadata.created_on))
+                                break
+                        if node_created_on is not None:
+                            break
+
+                    if node_created_on is None:
+                        raise Exception(
+                            'Node was not created.'
+                        )
+
+                    node_published_on = story_published_on if (
+                        story_published_on is not None and
+                        node_created_on is not None and
+                        story_published_on > node_created_on) else (
+                            node_created_on)
+                    node['first_publication_date_msecs'] = (
+                        node_published_on if node['status'] == 'Published'
+                        else None)
+                    node['planned_publication_date_msecs'] = (
+                        node['first_publication_date_msecs'])
+                    if node.get('last_modified_msecs') is None:
+                        node['last_modified_msecs'] = node_published_on
+
+        except Exception as e:
+            logging.exception(e)
+            return result.Err((story_model.id, e))
+        return result.Ok(story_model)
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+
+        updated_story_models_results = (
+            self.pipeline
+            | 'Get all story models' >> ndb_io.GetModels(
+                story_models.StoryModel.get_all())
+            | 'Update story node fields' >> beam.Map(self._update_story_node)
+        )
+
```

Comment:
(Curious) I've reviewed a similar job PR recently, whether it was your PR? (not sure!)

Reply:
Yes you have reviewed this PR once, but you did not point any detailed corrections back then.

Reply:
Can you please share the link to the PR?

Reply:
No, I was saying that you have reviewed this very PR in the past, not any other PR of mine.